koku/api/provider/serializers.py

```diff
@@ -288,9 +290,11 @@ def create(self, validated_data):
         credentials = authentication.get('credentials')
         provider_resource_name = credentials.get('provider_resource_name')
         provider_type = validated_data['type']
+        provider_type = Provider.PROVIDER_CASE_MAPPING.get(provider_type, provider_type)
```

Comment:
Should this be?
`provider_type = Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower())`

and then throw a validation error if provider type is None?